File: RepoTransBench/target_projects/Python/Java/initstring_linkedin2username/src/test/java/com/initstring/original/TestNamemutator.java
package com.initstring.original;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.MethodSource;
import java.util.*;
import java.util.stream.Stream;

import com.initstring.linkedin2username.NameMutator;

public class TestNamemutator {

    private static Stream<org.junit.jupiter.params.provider.Arguments> cleanAndSplitNameProvider() {
        return Stream.of(
            org.junit.jupiter.params.provider.Arguments.of("John Smith", Map.of("first", "john", "last", "smith", "second", "")),
            org.junit.jupiter.params.provider.Arguments.of("Jane D'oe", Map.of("first", "jane", "last", "doe", "second", "")),
            org.junit.jupiter.params.provider.Arguments.of("Dr. Ángela Gómez (MBA, PhD)", Map.of("first", "angela", "last", "gomez", "second", "")),
            org.junit.jupiter.params.provider.Arguments.of("José Niño", Map.of("first", "jose", "last", "nino", "second", "")),
            org.junit.jupiter.params.provider.Arguments.of("Joe (CTO) Bloggs", Map.of("first", "joe", "last", "bloggs", "second", ""))
        );
    }

    @ParameterizedTest
    @MethodSource("cleanAndSplitNameProvider")
    public void testCleanAndSplitName(String input, Map<String, String> expected) {
        NameMutator nm = new NameMutator(input);
        assertEquals(expected, nm.name);
    }
}

File: RepoTransBench/target_projects/Python/Rust/initstring_linkedin2username/tests/original/test_namemutator.rs
use std::collections::HashMap;
use linkedin2username::NameMutator;

fn mkmap(first: &str, last: &str, second: &str) -> HashMap<String, String> {
    let mut h = HashMap::new();
    h.insert("first".to_string(), first.to_string());
    h.insert("last".to_string(), last.to_string());
    h.insert("second".to_string(), second.to_string());
    h
}

#[test]
fn test_clean_and_split_name() {
    let samples = vec![
        ("John Smith", mkmap("john", "smith", "")),
        ("Jane D'oe", mkmap("jane", "doe", "")),
        ("Dr. Ángela Gómez (MBA, PhD)", mkmap("angela", "gomez", "")),
        ("José Niño", mkmap("jose", "nino", "")),
        ("Joe (CTO) Bloggs", mkmap("joe", "bloggs", "")),
    ];
    for (input, expected) in samples {
        let nm = NameMutator::new(input);
        assert_eq!(nm.name, expected);
    }
}